### PR TITLE
Message completion updated remark + exception message

### LINKED ISSUE #286: CompleteAsync() remarks should mention receiver affinity when completing messages
Similar to the full framework client, [`CompleteAsync`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.messagereceiver.completeasync) operation's remark should include similar warning:

>  If using AMQP, this operation can only be performed on messages that were received by this receiver.

[Current implementation](https://github.com/Azure/azure-service-bus-dotnet/blob/b39ef8a63c84d0b1bfa7c36ff13e2dc021b97833/src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs#L369-L416) doesn't have that remark.

### PR DESCRIPTION
**Pull request by SeanFeldman**
## Description

Fixes #286 

<details><summary>
This following list is used to make sure that common guidelines for a pull request are followed.
</summary>
- **Read the [contribution guidelines](./CONTRIBUTING.md).** 
- Make title of the pull request clear and informative. 
- Link to appropriate [issue](https://github.com/Azure/azure-service-bus-dotnet/issues).
- Include test coverage for the changes.
</details>

### CODE REVIEW

File: src/Microsoft.Azure.ServiceBus/Core/IMessageReceiver.cs [hunk 1]
```diff
@@ -97,6 +97,7 @@ public interface IMessageReceiver : IReceiverClient
         /// only when <see cref="ReceiveMode"/> is set to <see cref="ServiceBus.ReceiveMode.PeekLock"/>.
         /// </remarks>
         /// <param name="lockTokens">An <see cref="IEnumerable{T}"/> containing the lock tokens of the corresponding messages to complete.</param>
+        /// When using AMQP, this operation can only be performed on messages that were received by this receiver.
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Core/IMessageReceiver.cs]: We may not need to mention `When using AMQP`. That's the only one supported right now

File: src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs [hunk 2]
```diff
@@ -373,6 +373,7 @@ public Task<IList<Message>> ReceiveAsync(int maxMessageCount)
         /// <remarks>
         /// A lock token can be found in <see cref="Message.SystemPropertiesCollection.LockToken"/>, 
         /// only when <see cref="ReceiveMode"/> is set to <see cref="ServiceBus.ReceiveMode.PeekLock"/>.
+        /// When using AMQP, this operation can only be performed on messages that were received by this receiver.
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs]: Same as above

File: src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs [hunk 3]
```diff
@@ -386,6 +387,7 @@ public Task CompleteAsync(string lockToken)
...
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs]: Could you take care of this everywhere?

**Comment by nemakam** [comment]
And could you also update the remarks in `IReceiverClient.CompleteAsync()`?

**Comment by nemakam** [comment]
Now to think of it, this applies for all the methods which take locktoken. `Complete`, `Defer`, `Abandon`, `Deadletter`. :)

**Comment by SeanFeldman** [comment]
Wonder if remark is the right approach.

**Comment by SeanFeldman** [comment]
@nemakam incorporated your feedback.

File: src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs [hunk 4]
```diff
@@ -559,7 +566,7 @@ public Task CompleteAsync(string lockToken)
         /// <remarks>
         /// The first call to <see cref="PeekAsync()"/> fetches the first active message for this receiver. Each subsequent call 
         /// fetches the subsequent message in the entity.
-        /// Unliked a received messaged, peeked message will not have lock token associated with it, and hence it cannot be Completed/Abandoned/Defered/Deadlettered/Renewed.
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs]: >Unliked a received messaged [](start = 12, length = 27)

:D

**Pull request merged**